### PR TITLE
Fix typos in code

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix some typos in code.

Detailed description / Documentation draft:
It's easy:
```
milovidov@milovidov-desktop:~/work/ClickHouse/src$ find . -name '*.h' -or -name '*.cpp' | grep -v generated | xargs grep -n -o -P '[A-Z][a-z]{2,}|[a-z]{3,}' | clickhouse-local --structure 'file String, line UInt32, word String' --input-format CSV --format_csv_delimiter ':' --query "SELECT k, c, neighbor(k, 1) AS k2, ngramDistance(k, k2) FROM (SELECT lower(word) AS k, count() AS c, anyHeavy(file) AS file FROM table GROUP BY k ORDER BY k ASC, c DESC) WHERE c = 1 AND ngramDistance(k, k2) * least(length(k), length(k2)) <= 2 FORMAT PrettyCompactNoEscapes" | less
```

### CODE REVIEW

File: src/DataTypes/DataTypeCustomIPv4AndIPv6.cpp [hunk 1]
```diff
@@ -112,6 +112,10 @@ void registerDataTypeDomainIPv4AndIPv6(DataTypeFactory & factory)
         return std::make_pair(DataTypeFactory::instance().get("FixedString(16)"),
                               std::make_unique<DataTypeCustomDesc>(std::make_unique<DataTypeCustomFixedName>("IPv6"), std::make_unique<DataTypeCustomIPv6Serialization>()));
     });
+
+    /// MySQL, MariaDB
```
Comment (alexey-milovidov) [inline src/DataTypes/DataTypeCustomIPv4AndIPv6.cpp:116]: This is a bonus.

**Comment by alexey-milovidov** [comment]
> Integration tests

Broken in #11903

**Comment by alexey-milovidov** [comment]
@akuzm Any ideas on how to automate this?
I have tried `ispell` or `aspell` but it has too many false positives.

**Comment by akuzm** [comment]
> @akuzm Any ideas on how to automate this?
> I have tried `ispell` or `aspell` but it has too many false positives.

Not sure. The problem is that we don't want to support custom dictionaries? Maybe we can pipe all sources to clickhouse-local, build a list of words that are used in sources, and warn if we add a word that is not in the list, but is trigram-similar to an existing word, and detect typos in this way.
We could use postgres with ispell dictionaries, function ts_lexize(), to spellcheck the existing word.

**Comment by alexey-milovidov** [comment]
Yes, I had the same idea. We can support custom dictionary and store it directly in repository. But it will be very fragile if we don't normalize words. We can invest into text processing functions in ClickHouse but it's a big project.

**Comment by filimonov** [comment]
```
$ pip3 install codespell
$ pwd
/home/mfilimonov/workspace/ClickHouse/src

$ codespell

./Common/Volnitsky.h:347: occurence ==> occurrence
./Common/Volnitsky.h:536: ans ==> and
./Common/Volnitsky.h:540: ans ==> and
./Common/Volnitsky.h:540: ans ==> and
./Common/Volnitsky.h:556: ans ==> and
./Common/Volnitsky.h:556: ans ==> and
./Common/Volnitsky.h:563: ans ==> and
./Common/Volnitsky.h:566: ans ==> and
./Common/Volnitsky.h:574: ans ==> and
./Common/Volnitsky.h:578: ans ==> and
./Common/Volnitsky.h:578: ans ==> and
./Common/Volnitsky.h:594: ans ==> and
./Common/Volnitsky.h:594: ans ==> and
./Common/Volnitsky.h:599: ans ==> and
./Common/Volnitsky.h:601: ans ==> and
./Common/Volnitsky.h:605: ans ==> and
./Common/Volnitsky.h:612: ans ==> and
./Common/Volnitsky.h:628: ans ==> and
./Common/Volnitsky.h:629: ans ==> and
./Common/CurrentMetrics.cpp:36: backround ==> background
./Common/CurrentMetrics.cpp:37: backround ==> background
./Common/StackTrace.cpp:70: Non-existant ==> Non-existent
./Common/formatIPv6.h:35: atleast ==> at least
./Common/formatIPv6.h:71: atleast ==> at least
./Common/formatIPv6.h:180: inout ==> input, in out
./Common/CurrentThread.h:35: initializaiton ==> initialization
./Common/DNSResolver.cpp:285: becouse ==> because
./Common/Exception.cpp:121: creat ==> create
./Common/LRUCache.h:70: occuring ==> occurring
./Common/FieldVisitors.h:99: numberic ==> numeric
./Common/intExp.h:17: overlow ==> overflow
./Common/parseGlobs.cpp:12: fo ==> of, for
./Common/NamePrompter.h:80: ans ==> and
./Common/NamePrompter.h:81: ans ==> and
./Common/NamePrompter.h:86: ans ==> and
./Common/NamePrompter.h:88: ans ==> and
./Common/NamePrompter.h:88: ans ==> and
./Common/NamePrompter.h:89: ans ==> and
./Common/UTF8Helpers.cpp:98: Calcualte ==> Calculate
./Common/ArrayCache.h:560: occuring ==> occurring
./Common/ObjectPool.h:27: avaiable ==> available
./Common/RWLock.cpp:241: theses ==> these, thesis
./Common/SensitiveDataMasker.h:17: singelton ==> singleton
./Common/SensitiveDataMasker.h:23: singelton ==> singleton
./Common/SensitiveDataMasker.h:25: syncronization ==> synchronization
./Common/TaskStatsInfoGetter.h:21: capabilties ==> capabilities
./Common/SymbolIndex.cpp:35: splitted ==> split
./Common/SymbolIndex.cpp:37: splitted ==> split
./Common/SymbolIndex.cpp:311: splitted ==> split
./Common/QueryProfiler.h:28: Desctructor ==> Destructor
./Common/AutoArray.h:18: unchangable ==> unchangeable
./Common/PODArray.h:38: accidential ==> accidental
./Common/CurrentMetrics.h:55: amout ==> amount
./Common/CounterInFile.h:66: manulally ==> manually
./Common/Config/ConfigReloader.h:45: backround ==> background
./Common/Config/ConfigReloader.cpp:119: sucessfully ==> successfully
./Common/HashTable/FixedHashTable.h:32: accomodate ==> accommodate
./Common/HashTable/FixedHashTable.h:54: comparision ==> comparison
./Common/StringUtils/StringUtils.h:57: nd ==> and, 2nd
./Common/tests/chaos_sanitizer.cpp:13: Prooves ==> Proves
./Common/ZooKeeper/ZooKeeperImpl.cpp:116: Arbitary ==> Arbitrary
./Common/ZooKeeper/ZooKeeper.h:192: speficied ==> specified
./Access/AllowedClientHosts.cpp:159: expession ==> expression
./Access/IAccessStorage.h:37: Searchs ==> Searches
./Access/IAccessStorage.h:48: Searchs ==> Searches
./Processors/ConcatProcessor.h:9: arbitary ==> arbitrary
./Processors/IProcessor.h:71: arbitary ==> arbitrary
./Processors/IProcessor.h:71: arbitary ==> arbitrary
./Processors/IProcessor.h:72: choosed ==> chose, chosen
./Processors/IProcessor.h:74: arbitary ==> arbitrary
./Processors/IProcessor.h:75: arbitary ==> arbitrary
./Processors/IProcessor.h:105: arbitary ==> arbitrary
./Processors/ResizeProcessor.h:10: arbitary ==> arbitrary
./Processors/ResizeProcessor.h:10: arbitary ==> arbitrary
./Processors/ResizeProcessor.h:13: arbitary ==> arbitrary
./Processors/ResizeProcessor.h:13: arbitary ==> arbitrary
./Processors/Port.h:120: assigment ==> assignment
./Processors/QueryPipeline.cpp:102: fro ==> for, from
./Processors/Formats/IOutputFormat.h:37: intented ==> intended, indented
./Processors/Formats/Impl/AvroRowInputFormat.cpp:632: Succesfully ==> Successfully
./Processors/Formats/Impl/ArrowColumnToCHColumn.cpp:52: convertable ==> convertible
./Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp:45: splitted ==> split
./Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp:46: splitted ==> split
./Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp:48: splitted ==> split
./Processors/Transforms/FillingTransform.h:10: deafult ==> default
./Processors/Transforms/MergingAggregatedMemoryEfficientTransform.h:23: splitted ==> split
./Processors/Transforms/MergingAggregatedMemoryEfficientTransform.h:32: splitted ==> split
./Processors/Transforms/MergingAggregatedMemoryEfficientTransform.h:35: splitted ==> split
./Processors/Transforms/MergingAggregatedMemoryEfficientTransform.h:44: splitted ==> split
./Processors/Executors/PipelineExecutingBlockInputStream.h:38: singe ==> single
./Processors/Executors/ThreadsQueue.h:12: Oll ==> All, ole, old, olly, oil
./TableFunctions/ITableFunctionFileLike.h:11: formated ==> formatted
./IO/ReadHelpers.h:673: ommiting ==> omitting
./IO/WriteBufferFromHTTPServerResponse.h:100: progess ==> progress
./IO/tests/var_uint.cpp:27: sucessfully ==> successfully
./Columns/IColumn.h:228: sortings ==> sorting
./Columns/IColumn.h:239: sortings ==> sorting
./Server/PostgreSQLHandler.cpp:83: exteneded ==> extended
./Client/TimeoutSetter.cpp:37: catched ==> caught
./Databases/DatabaseOnDisk.cpp:426: loger ==> logger, lodger, longer
./Databases/DatabaseOnDisk.cpp:448: loger ==> logger, lodger, longer
./Databases/DatabaseOnDisk.cpp:472: loger ==> logger, lodger, longer
./Functions/FunctionBinaryArithmetic.h:352: convertion ==> conversion
./Functions/formatString.h:228: ans ==> and
./Functions/formatString.h:239: perfomance ==> performance
./Functions/formatDateTime.cpp:54: charaters ==> characters
./Functions/pointInPolygon.cpp:418: parseRing ==> parsing
./Functions/pointInPolygon.cpp:440: parseRing ==> parsing
./Functions/pointInPolygon.cpp:461: parseRing ==> parsing
./Functions/pointInPolygon.cpp:466: parseRing ==> parsing
./Functions/FunctionsHashing.h:350: singed ==> signed, singled
./Functions/FunctionsHashing.h:485: non-reproducable ==> non-reproducible
./Functions/FunctionsHashing.h:508: non-reproducable ==> non-reproducible
./Functions/FunctionsStringSimilarity.cpp:141: cyrrilic ==> cyrillic
./Functions/PolygonUtils.h:182: splitted ==> split
./Functions/PolygonUtils.h:183: splitted ==> split
./Functions/PolygonUtils.h:184: splitted ==> split
./Functions/geohashesInBox.cpp:136: Arrary ==> Array
./Functions/extractAllGroups.h:34: grop ==> group, drop
./Functions/FunctionCustomWeekToSomething.h:54: 3nd ==> 3rd
./Functions/FunctionCustomWeekToSomething.h:70: 3nd ==> 3rd
./Functions/FunctionCustomWeekToSomething.h:78: 3nd ==> 3rd
./Functions/isValidUTF8.cpp:233: caluclated ==> calculated
./Functions/FunctionsBitmap.h:36: Retrun ==> Return
./Functions/FunctionsBitmap.h:39: Retrun ==> Return
./Functions/FunctionsBitmap.h:63: Retrun ==> Return
./Functions/FunctionDateOrDateTimeAddInterval.h:423: INTPUT ==> INPUT
./Functions/multiIf.cpp:52: thenN ==> then
./Functions/globalVariable.cpp:46: Agrument ==> Argument
./Functions/Regexps.h:142: accrording ==> according
./Functions/FunctionsConversion.h:381: Arbitary ==> Arbitrary
./Functions/FunctionsConversion.h:581: resultion ==> resolution
./Functions/FunctionsConversion.h:963: Someting ==> Something
./Functions/FunctionsConversion.h:1207: Someting ==> Something
./Functions/FunctionsConversion.h:1376: shrinked ==> shrunk
./Functions/URL/tldLookup.gperf:199: ba ==> by, be
./Functions/URL/tldLookup.gperf:200: ba ==> by, be
./Functions/URL/tldLookup.gperf:201: ba ==> by, be
./Functions/URL/tldLookup.gperf:202: ba ==> by, be
./Functions/URL/tldLookup.gperf:203: ba ==> by, be
./Functions/URL/tldLookup.gperf:204: ba ==> by, be
./Functions/URL/tldLookup.gperf:367: fot ==> for
./Functions/URL/tldLookup.gperf:383: jus ==> just
./Functions/URL/tldLookup.gperf:495: presse ==> pressed, press
./Functions/URL/tldLookup.gperf:1036: ba ==> by, be
./Functions/URL/tldLookup.gperf:1191: ot ==> to, of, or
./Functions/URL/tldLookup.gperf:1256: te ==> the, be
./Functions/URL/tldLookup.gperf:1449: presse ==> pressed, press
./Functions/URL/tldLookup.gperf:1617: presse ==> pressed, press
./Functions/URL/tldLookup.gperf:1932: linz ==> lines
./Functions/URL/tldLookup.gperf:2301: herad ==> heard, hera
./Functions/URL/tldLookup.gperf:2823: rade ==> read, raid
./Functions/URL/tldLookup.gperf:2877: stange ==> strange
./Functions/URL/tldLookup.gperf:3618: od ==> of
./Functions/URL/tldLookup.gperf:3631: te ==> the, be
./Functions/URL/tldLookup.gperf:3699: nd ==> and, 2nd
./Functions/URL/tldLookup.gperf:4408: ba ==> by, be
./Functions/URL/tldLookup.generated.cpp:3807: jus ==> just
./Functions/URL/tldLookup.generated.cpp:4581: nd ==> and, 2nd
./Functions/URL/tldLookup.generated.cpp:4682: ba ==> by, be
./Functions/URL/tldLookup.generated.cpp:4732: ba ==> by, be
./Functions/URL/tldLookup.generated.cpp:5472: ba ==> by, be
./Functions/URL/tldLookup.generated.cpp:7702: ba ==> by, be
./Functions/URL/tldLookup.generated.cpp:8457: fot ==> for
./Functions/URL/tldLookup.generated.cpp:14871: ba ==> by, be
./Functions/URL/tldLookup.generated.cpp:20858: stange ==> strange
./Functions/URL/tldLookup.generated.cpp:21509: od ==> of
./Functions/URL/tldLookup.generated.cpp:23465: ba ==> by, be
./Functions/URL/tldLookup.generated.cpp:24925: presse ==> pressed, press
./Functions/URL/tldLookup.generated.cpp:25289: ot ==> to, of, or
./Functions/URL/tldLookup.generated.cpp:28262: presse ==> pressed, press
./Functions/URL/tldLookup.generated.cpp:30605: rade ==> read, raid
./Functions/URL/tldLookup.generated.cpp:31136: ba ==> by, be
./Functions/URL/tldLookup.generated.cpp:35322: presse ==> pressed, press
./Functions/URL/tldLookup.generated.cpp:37249: ba ==> by, be
./Functions/URL/tldLookup.generated.cpp:38025: te ==> the, be
./Functions/URL/tldLookup.generated.cpp:41510: te ==> the, be
./Functions/URL/tldLookup.generated.cpp:45369: herad ==> heard, hera
./Functions/URL/tldLookup.generated.cpp:69752: linz ==> lines
./Functions/array/arrayEnumerateRanked.h:20: occurence ==> occurrence
./Functions/array/arrayEnumerateRanked.cpp:91: cant ==> can't
./Functions/array/arrayAUC.cpp:38: FPR ==> FOR, FAR, FPS
./Functions/array/arrayAUC.cpp:39: FPR ==> FOR, FAR, FPS
./Functions/array/arrayAUC.cpp:41: FPR ==> FOR, FAR, FPS
./Functions/array/arrayAUC.cpp:52: FPR ==> FOR, FAR, FPS
./Functions/array/arrayAUC.cpp:65: FPR ==> FOR, FAR, FPS
./Functions/array/arrayAUC.cpp:66: FPR ==> FOR, FAR, FPS
./Functions/array/arrayAUC.cpp:67: FPR ==> FOR, FAR, FPS
./Functions/array/arrayAUC.cpp:68: FPR ==> FOR, FAR, FPS
./Functions/array/arrayAUC.cpp:69: FPR ==> FOR, FAR, FPS
./Storages/IStorage.h:192: overriden ==> overridden
./Storages/ColumnDependency.h:12: depenendecies ==> dependencies
./Storages/StorageReplicatedMergeTree.h:369: metdata ==> metadata
./Storages/TTLDescription.h:46: Expresion ==> Expression
./Storages/IStorage.cpp:309: overriden ==> overridden
./Storages/CheckResults.h:12: simplier ==> simpler
./Storages/MutationCommands.h:20: statisfy ==> satisfy
./Storages/StorageReplicatedMergeTree.cpp:641: concurently ==> concurrently
./Storages/StorageReplicatedMergeTree.cpp:3233: attmept ==> attempt
./Storages/StorageReplicatedMergeTree.cpp:3718: successfull ==> successful
./Storages/KeyDescription.cpp:58: loosing ==> losing
./Storages/AlterCommands.h:143: dependecies ==> dependencies
./Storages/MergeTree/MergeTreeData.h:420: determing ==> determining, determine
./Storages/MergeTree/MergeTreeData.h:832: overriden ==> overridden
./Storages/MergeTree/MergeTreeData.h:832: childs ==> children, child's
WARNING: Binary file: ./Storages/MergeTree/MergeTreeIndexGranularityInfo.h.gch 
./Storages/MergeTree/MergeTreeDataSelectExecutor.cpp:907: splitted ==> split
./Storages/MergeTree/MergeTreeDataSelectExecutor.cpp:910: splitted ==> split
./Storages/MergeTree/MergeTreeDataSelectExecutor.cpp:917: splitted ==> split
./Storages/MergeTree/MergeTreeData.cpp:374: exsists ==> exists
./Storages/MergeTree/ReplicatedMergeTreeQueue.h:121: dowload ==> download
./Storages/MergeTree/ReplicatedMergeTreeQueue.h:360: Cheks ==> Checks
./Storages/MergeTree/IMergeTreeReader.cpp:163: wll ==> will
./Storages/MergeTree/MergeTreeIndexFullText.cpp:643: calcualted ==> calculated
./Storages/MergeTree/MergeTreeIndexFullText.cpp:664: statisfies ==> satisfies
./Storages/MergeTree/MergeTreeThreadSelectBlockInputProcessor.h:45: readed ==> read, readd, readded
./Storages/MergeTree/ReplicatedMergeTreeAltersSequence.h:11: depency ==> dependency
./Storages/MergeTree/ReplicatedMergeTreeQueue.cpp:990: alredy ==> already
./Storages/MergeTree/ReplicatedMergeTreeQueue.cpp:1238: ot ==> to, of, or
./Storages/MergeTree/ReplicatedMergeTreeQueue.cpp:1882: sence ==> sense, since
./Storages/MergeTree/ReplicatedMergeTreePartCheckThread.cpp:356: cheking ==> checking
./Storages/MergeTree/registerStorageMergeTree.cpp:522: explicitely ==> explicitly
./Storages/MergeTree/registerStorageMergeTree.cpp:591: explicitely ==> explicitly
./Storages/MergeTree/MergeTreeThreadSelectBlockInputProcessor.cpp:32: sence ==> sense, since
./Storages/MergeTree/IMergeTreeDataPartWriter.cpp:95: Implemetation ==> Implementation
./Storages/MergeTree/IMergeTreeDataPartWriter.cpp:95: splitted ==> split
./Storages/MergeTree/LevelMergeSelector.cpp:92: Minumum ==> Minimum
./Storages/MergeTree/MergeTreeRangeReader.cpp:881: shrinked ==> shrunk
./Storages/MergeTree/IMergeTreeDataPart.h:194: Commited ==> Committed
./Storages/MergeTree/IMergeTreeDataPart.h:359: files' ==> file's
./Storages/MergeTree/MergeTreePartsMover.h:57: commited ==> committed
./Storages/MergeTree/MergeTreeDataMergerMutator.cpp:508: Auxilliary ==> Auxiliary
./Storages/LiveView/LiveViewBlockInputStream.h:10: outputing ==> outputting
./Storages/LiveView/LiveViewBlockInputStream.h:130: availble ==> available
./Storages/LiveView/LiveViewEventsBlockInputStream.h:28: outputing ==> outputting
./Storages/LiveView/LiveViewEventsBlockInputStream.h:164: availble ==> available
./Storages/tests/gtest_SplitTokenExtractor.cpp:181: bounday ==> boundary
./Storages/Distributed/DistributedBlockOutputStream.h:32: splitted ==> split
./Storages/Kafka/ReadBufferFromKafkaConsumer.cpp:40: synchroniously ==> synchronously
./Storages/Kafka/ReadBufferFromKafkaConsumer.cpp:47: synchroniously ==> synchronously
./Storages/Kafka/ReadBufferFromKafkaConsumer.cpp:195: commited ==> committed
./Storages/Kafka/ReadBufferFromKafkaConsumer.cpp:197: commited ==> committed
./Storages/Kafka/ReadBufferFromKafkaConsumer.cpp:206: commited ==> committed
./Storages/Kafka/ReadBufferFromKafkaConsumer.cpp:216: commited ==> committed
./Storages/Kafka/ReadBufferFromKafkaConsumer.cpp:218: ot ==> to, of, or
./Storages/Kafka/ReadBufferFromKafkaConsumer.cpp:219: commited ==> committed
./Storages/Kafka/ReadBufferFromKafkaConsumer.cpp:350: commited ==> committed
./Storages/Kafka/ReadBufferFromKafkaConsumer.cpp:412: commiting ==> committing
./Storages/Kafka/parseSyslogLevel.cpp:4: sepearate ==> separate
./Storages/Kafka/WriteBufferToKafkaProducer.cpp:60: delimeter ==> delimiter
./Storages/System/StorageSystemContributors.generated.cpp:108: Colum ==> Column
./Storages/System/StorageSystemContributors.generated.cpp:232: Maks ==> Mask, masks, makes, make
./Compression/CompressionCodecDoubleDelta.cpp:155: unsinged ==> unsigned
./Compression/CompressionCodecDoubleDelta.cpp:293: constrast ==> contrast
./Compression/CompressionCodecDoubleDelta.h:14: contant ==> constant, content
./Compression/CompressionCodecDoubleDelta.h:48: ommited ==> omitted
./Compression/CompressedReadBufferBase.cpp:50: endianess ==> endianness
./Compression/tests/gtest_compressionCodec.cpp:549: iput ==> input
./Compression/tests/gtest_compressionCodec.cpp:1277: sence ==> sense, since
./Compression/tests/gtest_compressionCodec.cpp:1278: agains ==> against, again
./Parsers/ExpressionElementParsers.cpp:1137: FRO ==> FOR, FROM
./Parsers/ExpressionElementParsers.cpp:1137: FRO ==> FOR, FROM
./Parsers/ASTLiteral.cpp:20: rearely ==> rarely
./Parsers/ParserCreateQuery.h:357: DICTIONAY ==> DICTIONARY
./Parsers/ExpressionElementParsers.h:353: alowed ==> allowed
./Parsers/ASTDictionaryAttributeDeclaration.h:21: atribute ==> attribute
./Interpreters/IdentifierSemantic.h:19: calss ==> calls, class
./Interpreters/MergeJoin.cpp:447: splitted ==> split
./Interpreters/HashJoin.cpp:578: Noone ==> No one
./Interpreters/MutationsInterpreter.cpp:465: colums ==> columns
./Interpreters/JoinedTables.h:18: occurance ==> occurrence
./Interpreters/JoinedTables.h:23: contex ==> context
./Interpreters/HashJoin.h:91: talbe ==> table
./Interpreters/InterpreterAlterQuery.cpp:86: asyncronously ==> asynchronously
./Interpreters/MergeJoin.h:42: prefered ==> preferred
./Interpreters/ArrayJoinedColumnsVisitor.h:101: splitted ==> split
./Interpreters/ArrayJoinedColumnsVisitor.h:108: splitted ==> split
./Interpreters/ArrayJoinedColumnsVisitor.h:108: splitted ==> split
./Interpreters/ArrayJoinedColumnsVisitor.h:112: splitted ==> split
./Interpreters/ArrayJoinedColumnsVisitor.h:112: splitted ==> split
./Interpreters/ArrayJoinedColumnsVisitor.h:122: splitted ==> split
./Interpreters/ArrayJoinedColumnsVisitor.h:122: splitted ==> split
./Interpreters/ArrayJoinedColumnsVisitor.h:127: splitted ==> split
./Interpreters/ArrayJoinedColumnsVisitor.h:127: splitted ==> split
./Interpreters/DDLWorker.cpp:690: occured ==> occurred
./Interpreters/DDLWorker.cpp:904: childs ==> children, child's
./Interpreters/DDLWorker.cpp:905: childs ==> children, child's
./Interpreters/DDLWorker.cpp:922: occured ==> occurred
./Interpreters/Context.h:532: opertaions ==> operations
./Interpreters/Context.h:533: cheks ==> checks
./Interpreters/InJoinSubqueriesPreprocessor.h:44: overriden ==> overridden
./Interpreters/Context.cpp:1795: occured ==> occurred
./Interpreters/Context.cpp:1938: beacuse ==> because
./Interpreters/ArithmeticOperationsInAgrFuncOptimize.cpp:140: ans ==> and
./Interpreters/ArithmeticOperationsInAgrFuncOptimize.cpp:141: ans ==> and
./Interpreters/ArithmeticOperationsInAgrFuncOptimize.cpp:142: ans ==> and
./Interpreters/ArithmeticOperationsInAgrFuncOptimize.cpp:145: ans ==> and
./Interpreters/ArithmeticOperationsInAgrFuncOptimize.cpp:146: ans ==> and
./Interpreters/ArithmeticOperationsInAgrFuncOptimize.cpp:147: ans ==> and
./Interpreters/ArithmeticOperationsInAgrFuncOptimize.cpp:156: ans ==> and
./Interpreters/ArithmeticOperationsInAgrFuncOptimize.cpp:157: ans ==> and
./Interpreters/ArithmeticOperationsInAgrFuncOptimize.cpp:158: ans ==> and
./Interpreters/ExternalLoaderDatabaseConfigRepository.h:11: existance ==> existence
./Interpreters/JoinToSubqueryTransformVisitor.h:14: rewriten ==> rewritten
./Interpreters/Set.cpp:583: layed ==> laid
./Interpreters/ExternalLoaderXMLConfigRepository.h:38: dicrectory ==> directory
./Interpreters/SyntaxAnalyzer.cpp:604: splitted ==> split
./Interpreters/SyntaxAnalyzer.cpp:605: splitted ==> split
./Interpreters/SyntaxAnalyzer.cpp:605: splitted ==> split
./Interpreters/SyntaxAnalyzer.cpp:607: splitted ==> split
./Interpreters/InterpreterCreateQuery.cpp:298: Cant ==> Can't
./Interpreters/InterpreterCreateQuery.cpp:640: doesnt ==> doesn't, does not
./Interpreters/ExpressionActions.h:141: splitted ==> split
./Interpreters/tests/internal_iotop.cpp:57: intitial ==> initial
./AggregateFunctions/AggregateFunctionWindowFunnel.h:151: statisfied ==> satisfied
./AggregateFunctions/AggregateFunctionWindowFunnel.h:163: withing ==> within
./AggregateFunctions/IAggregateFunctionCombinator.h:57: agggregate ==> aggregate
./AggregateFunctions/AggregateFunctionSequenceMatch.h:264: preceeded ==> preceded
./AggregateFunctions/AggregateFunctionSumMap.cpp:127: argumens ==> arguments
./AggregateFunctions/AggregateFunctionSumMap.cpp:136: argumens ==> arguments
./AggregateFunctions/AggregateFunctionNull.h:31: presense ==> presence
./AggregateFunctions/AggregateFunctionGroupBitmapData.h:12: inteface ==> interface
./AggregateFunctions/AggregateFunctionGroupBitmapData.h:512: ans ==> and
./AggregateFunctions/AggregateFunctionGroupBitmapData.h:518: ans ==> and
./AggregateFunctions/AggregateFunctionGroupBitmapData.h:521: ans ==> and
./AggregateFunctions/AggregateFunctionGroupBitmapData.h:521: ans ==> and
./AggregateFunctions/AggregateFunctionGroupBitmapData.h:522: ans ==> and
./AggregateFunctions/AggregateFunctionGroupBitmapData.h:523: ans ==> and
./AggregateFunctions/AggregateFunctionGroupBitmapData.h:525: ans ==> and
./AggregateFunctions/AggregateFunctionMLMethod.cpp:203: sence ==> sense, since
./DataTypes/NestedUtils.cpp:76: splitted ==> split
./DataTypes/NestedUtils.cpp:77: splitted ==> split
./DataTypes/NestedUtils.cpp:142: splitted ==> split
./DataTypes/NestedUtils.cpp:143: splitted ==> split
./DataTypes/NestedUtils.cpp:145: splitted ==> split
./DataTypes/NestedUtils.cpp:145: splitted ==> split
./DataTypes/NestedUtils.cpp:176: splitted ==> split
./DataTypes/NestedUtils.cpp:179: splitted ==> split
./DataTypes/NestedUtils.cpp:181: splitted ==> split
./DataTypes/NestedUtils.cpp:192: splitted ==> split
./DataTypes/DataTypeCustomSimpleAggregateFunction.h:24: instanciated ==> instantiated
./DataTypes/DataTypeDateTime64.h:60: fucntion ==> function
./DataTypes/DataTypeDateTime64.h:65: Wehere ==> Where
./DataTypes/IDataType.h:310: overriden ==> overridden
./DataTypes/getLeastSupertype.cpp:283: convertion ==> conversion
./Core/Settings.h:168: preformed ==> performed
./Core/DecimalFunctions.h:34: negatve ==> negative
./Core/BlockInfo.h:48: extention ==> extension
./Core/Defines.h:67: Mininum ==> Minimum
./Core/tests/gtest_DecimalFunctions.cpp:59: unfortunatelly ==> unfortunately
./DataStreams/AsynchronousBlockInputStream.h:86: backgroud ==> background
./DataStreams/ParallelParsingBlockInputStream.h:121: should't ==> shouldn't
./DataStreams/MergingSortedBlockInputStream.cpp:252: childs ==> children, child's
./DataStreams/MergingSortedBlockInputStream.h:24: readed ==> read, readd, readded
./DataStreams/AddingDefaultsBlockInputStream.cpp:48: Mismach ==> Mismatch
./DataStreams/AddingDefaultsBlockInputStream.cpp:154: alredy ==> already
./DataStreams/PushingToViewsBlockOutputStream.cpp:32: deduplactes ==> deduplicates
./DataStreams/ParallelInputsProcessor.h:330: consequentially ==> consequently
./DataStreams/ParallelParsingBlockInputStream.cpp:76: occured ==> occurred
./Dictionaries/ComplexKeyCacheDictionary.h:690: overlaped ==> overlapped
./Dictionaries/CacheDictionary.h:327: overlaped ==> overlapped
./Dictionaries/CacheDictionary.h:362: retreived ==> retrieved
./Dictionaries/CacheDictionary.h:386: aquiring ==> acquiring
```

There are false matches (but not too much), and be adjusted by flags / dicts. Check:  https://github.com/codespell-project/codespell

**Comment by filimonov** [comment]
There is also 
```
pip3 install scspell3k 

find . -name '*.h' -or -name '*.cpp' | xargs scspell --report-only --use-builtin-base-dict ./
```
But it require tuning, storng & filling dictionaries incrementally etc.

**Comment by alexey-milovidov** [comment]
https://github.com/ClickHouse/ClickHouse/pull/12003#issuecomment-651203701

This result looks very promising, we can incorporate this tool with a simple exception list.

**Comment by filimonov** [comment]
It also have plenty of options / switches

**Comment by alexey-milovidov** [comment]
#13511
#13513

**Pull request merged**